src/ompl/geometric/planners/bitstar/BITstar.h

```diff
@@ -0,0 +1,614 @@
+            unsigned int                                             numVerticesDisconnected_;
+
+            /** \brief The number of times a state in the graph was rewired. Accessible via rewiringProgressProperty */
+            unsigned int                                             numRewirings_;
+
+            /** \brief The number of state collision checks. Accessible via stateCollisionCheckProgressProperty */
+            unsigned int                                             numStateCollisionChecks_;
+
+            /** \brief The number of edge collision checks. Accessible via edgeCollisionCheckProgressProperty */
+            unsigned int                                             numEdgeCollisionChecks_;
+
+            /** \brief The number of nearest neighbour calls. Accessible via nearestNeighbourProgressProperty */
+            unsigned int                                             numNearestNeighbours_;
+            ///////////////////////////////////////
+
+            ///////////////////////////////////////
+            //Parameters - Set defaults in construction/setup and DO NOT reset in clear.
+            /** \brief Whether to use a strict-queue ordering (param) */
+            bool                                                     useStrictQueueOrdering_;
+
+            /** \brief The rewiring factor, s, so that r_rrg = s \times r_rrg* > r_rrg* (param) */
+            double                                                   rewireFactor_;
+
+            /** \brief The number of samples per batch (param) */
+            unsigned int                                             samplesPerBatch_;
+
+            /** \brief Track edges that have been checked and failed so they never reenter the queue. (param) */
+            bool                                                     useFailureTracking_;
+
+            /** \brief Option to use k-nearest search for rewiring (param) */
+            bool                                                     useKNearest_;
+
+            /** \brief Whether to use graph pruning (param) */
+            bool                                                     usePruning_;
+
+            /** \brief The fractional decrease in solution cost required to trigger pruning (param) */
+            double                                                   pruneFraction_;
+
+            /** \brief Whether to delay rewiring until a solution is found (param) */
+            bool                                                     delayRewiring_;
+
+            /** \brief Whether to stop the planner as soon as the path changes (param) */
+            bool                                                     stopOnSolnChange_;
+            ///////////////////////////////////////
+        }; //class: BITstar
+    } //geometric
+} //ompl
+
+
+//BIT* Includes:
+//The Vertex ID generator class
+#include "ompl/geometric/planners/bitstar/datastructures/IdGenerator.h"
```

Comment:
I wish I could... The issue is that these three classes are defined as *member* classes of the `BITstar` class. If they're included before it is defined, the keyword `BITstar` will be unknown... When I tried to forward declare `BITstar` in the member classes, all hell broke loose, and I really didn't want to define these classes inline in this file (like some other planners do with their `Motion` class)...

There is a comment at the top of the `BITstar.h` explaining why three more includes are at the bottom... I figured it sort of mirrored what's done with the implementation file for a templated class...